node/src/components/console/tasks.rs

```diff
@@ -0,0 +1,335 @@
+        }
+    }
+
+    /// Constructs a new failed outcome.
+    fn failed<S: ToString>(reason: S) -> Self {
+        Outcome::Failure {
+            reason: reason.to_string(),
+        }
+    }
+}
+
+impl Display for Outcome {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Outcome::Success { msg } => {
+                write!(f, "OK   {}", msg)
+            }
+            Outcome::Failure { reason } => {
+                write!(f, "ERR  {}", reason)
+            }
+        }
+    }
+}
+
+/// Configuration for a connection console session.
+#[derive(Copy, Clone, Debug, Default, Serialize)]
+struct Session {
+    /// Whether or not to suppress the operation outcome.
+    quiet: bool,
+    /// Output format to send to client.
+    output: OutputFormat,
+}
+
+impl Display for Session {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl Session {
+    /// Creates a serializer for an `EraDump`.
+    fn create_era_dump_serializer(&self) -> fn(&EraDump<'_>) -> Result<Vec<u8>, Cow<'static, str>> {
+        // TODO: This function could probably be a generic serialization function for any `T`, but
+        // the conversion is tricky due to the lifetime arguments on `EraDump` and has not been done
+        // yet.
+        match self.output {
+            OutputFormat::Interactive => |data: &EraDump| {
+                let mut buf = data.to_string().into_bytes();
+                buf.push(b'\n');
+                Ok(buf)
+            },
+            OutputFormat::Json => |data: &EraDump| {
+                let mut buf = serde_json::to_vec(&data).map_err(|err| {
+                    Cow::Owned(format!("failed to serialize era dump as JSON: {}", err))
+                })?;
+                buf.push(b'\n');
+                Ok(buf)
+            },
+            OutputFormat::Bincode => |data: &EraDump| {
+                bincode::serialize(&data).map_err(|err| {
+                    Cow::Owned(format!("failed to serialize era dump as bincode: {}", err))
+                })
+            },
+        }
+    }
+
+    /// Processes a single command line sent from a client.
+    async fn process_line<REv>(
+        &mut self,
+        effect_builder: EffectBuilder<REv>,
+        writer: &mut OwnedWriteHalf,
+        line: &str,
+    ) -> io::Result<()>
+    where
+        REv: From<DumpConsensusStateRequest> + Send,
+    {
+        debug!(%line, "line received");
+        match Command::from_line(line) {
+            Ok(ref cmd) => {
+                info!(?cmd, "processing command");
+                match cmd.action {
+                    Action::Session => {
+                        self.send_outcome(writer, &Outcome::success("showing session info"))
+                            .await?;
+                        self.send_to_client(writer, &self).await?;
+                    }
+                    Action::Set { quiet, output } => {
```

Comment:
In order to match the flow in other two actions, there should be something like `self.send_outcome(writer, &Outcome::success("changing session"));` here.

Reply:
There is, in lines 150ff? Do you want this to happen earlier?

Reply:
That's not the same as in line 150 you're confirming the action and my understanding is that the first output sent is a confirmation of action being taken.